### PR TITLE
Add function to change song speed and update time information for waterfall

### PR DESCRIPTION
**Pull request by pohlinwei**
I added a helper function `changeSongSpeed` under `src/components/utils.ts` and replaced the hardcoded values for `bpm`, `beatsPerBar`, etc. with values from the JSON object.

### CODE REVIEW

File: src/api/genres.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import base from './base';
 
 const genresAPI = {
   getGenres: async (): Promise<[Genre]> => {
+    console.log('hello');
```
Comment (jonchan51) [inline src/api/genres.ts:6]:
```suggestion

```

File: src/components/DuetRoom.tsx [hunk 2]
```diff
@@ -142,8 +142,15 @@ const DuetRoom: React.FC<{ maybeRoomId: string | null; isCreate: boolean }> = ({
   const partsSelection = getPartsSelection(roomState);
   const myPart = getMyPart(roomState, playerId);
 
-  // Calculate keyboard dimension
+  // song information
   const tracks = chosenSongMIDI.tracks;
+  const bpm = chosenSongMIDI.header?.tempos[0].bpm;
```
Comment (jonchan51) [inline src/components/DuetRoom.tsx:147]: Can we move these variables into the if condition where waterfall is used, inside the `middleBox` function? I think it doesn't actually affect the correctness so it should be fine even if you don't though.

Also, what if bpm, beatsPerBar, etc. are missing? Or are they definitely included in every JSON file?

Comment (pohlinwei) [reply src/components/DuetRoom.tsx:147]: Sure! 

The information should be in every JSON file since we need it for the playback. 😅

**Review by Na-Nazhou — APPROVED** [review]

File: src/components/Waterfall/utils.ts [hunk 3]
```diff
@@ -17,15 +17,23 @@ export const convertTimeInfoToMilliseconds = (notes: Array<Note>) => {
...
  * Calculates the amount of time we should look ahead by in milliseconds.
  *
  * In this case, we try to look ahead by one musical bar.
  */
-export const calculateLookAheadTime = (bpm: number, beatsPerBar: number) => {
-  // TODO: support non-quarter notes (need to define new variable)
+export const calculateLookAheadTime = (
+  bpm: number,
+  beatsPerBar: number,
+  noteDivision: number
```
Comment (Na-Nazhou) [inline src/components/Waterfall/utils.ts:29]: how will the noteDivision be used? i think it is the state setter

**Pull request merged**